### ISSUE TITLE
Using any forwarder crashes software

### ISSUE THREAD

**Issue by Avellea** [XITRIX/Moonlight-Switch#179]
Hi there.

Attempting to use any forwarder created with NTON will fail to launch. Currently using the latest versions of NTON and Moonlight-Switch on 18.1.0 Atmosphere

The crash happens before I even see a UI, when the Switch logo is still spinning in the bottom right corner.

I have tested with both a blank forwarder to the application, and `--host=censored --appid=1093255277 --appname=Steam`

Forwarders I make for other applications such as Goldleaf and mGBA work without issue.

![image](https://github.com/user-attachments/assets/dfe8904e-3fe7-4d59-8cc3-22871e2f0270)
![image](https://github.com/user-attachments/assets/826121b5-ddb1-455f-9090-6c137b93c69a)

**Comment by XITRIX** [comment]
Previously there was someone who had the same issue, and the problem was that he wrote host in incorrect format, with --host="censored" I cannot confirm that you are writing everything correct, could you at least replacing some characters show how you white it?

**Comment by Avellea** [comment]
> could you at least replacing some characters show how you white it?

Sure, it ended up being formatted as `ab:cd:ef:01:23:45`

> Ah, wait, you have crash even with blank forwarder? That's weird ...

Could this possibly be due to firmware 18.1.0? I made sure to redump my prod.keys, as well as make sure all sigpatches and sys-patch were updated.

**Comment by XITRIX** [comment]
Not sure to be honest, I still not updated from 17.x

**Comment by Avellea** [comment]
Understandable. I didn't have moonlight installed to test on a previous firmware sadly, so that's still up in the air.

Is there any way to get a crash log of some sort for you? I haven't touched the development side of the Switch nearly as much as I have the PSVita.

**Comment by XITRIX** [comment]
Crash logs from Switch was always a problem for me, I developed the app on Mac, tested in xCode and launched on Switch as it is, debugging Switch specific issues with console logs :D

**Comment by Avellea** [comment]
So what exactly can be done?

**Comment by XITRIX** [comment]
Let me try to create forwarder on my device, I'll double check if it works or not.
Also I could try to release my own tool to generate forwarder, which thought to implement inside Moonlight-Switch itself, but Switch failed to generate working forwarders probably because of incorrectly working SSL lib

**Comment by Avellea** [comment]
Sounds good. I'm available to provide any information or test anything if needed. Thanks for taking the time to look into this!

**Comment by XITRIX** [comment]
[Moonlight by XITRIX [01D434D98213E000][v0].zip](https://github.com/user-attachments/files/16487551/Moonlight.by.XITRIX.01D434D98213E000.v0.zip)

Could you try to install this one? NRO should be in the path: `/switch/Moonlight-Switch/Moonlight-Switch.nro`
It should just launch the app, I checked, it works for me

**Comment by Avellea** [comment]
Loads right up without issue.

**Comment by XITRIX** [comment]
I created it with NTON, copy-pasted arguments from ReadMe, looks like it could be something wrong with your NTON installation or ... I don't know.

Also, if you cannot run forwarder without any arguments, so it's definitely something wrong with your NTON

**Comment by Avellea** [comment]
That's what I'm confused about... It's a fresh install of NTON (even redownloaded and tried again) and fresh dumped keys as well.

**Comment by XITRIX** [comment]
Do you click "Create ROM forwarder"?

**Comment by Avellea** [comment]
Build NSP Forwarder.

**Comment by XITRIX** [comment]
Btw, if my forwarder works for you, so it's definitely something wrong on your side, not sure how I could help you here

**Comment by Avellea** [comment]
I've checked for typos, incorrect paths, everything. It's _just_ Switch-Moonlight

I even recreated my mGBA forwarder to test. I'm not really sure what the issue could be at this point.

**Comment by XITRIX** [comment]
Once again, you cannot create forwarder even without arguments, it will still crash, yes?

**Comment by Avellea** [comment]
That is correct. That does not mean it is guaranteed to be something on my end though, as the exact same setup works for any other .nro I throw at it.

**Comment by XITRIX** [comment]
Just one thing, have no idea if it could be the reason, I've enabled video capture in my forwarder, in the app I also have a piece of code, that tries to enable it. MAYBE if forwarder not allows it (no idea how) and app tries to enable it with no result it crash, but it's just a pure theory, I did not tried it on practice, but it is the only possible difference between our forwarders

**Comment by Avellea** [comment]
I'll try this and let you know.

**Comment by maus-me** [comment]
I was similarly experiencing this issue with NTON today specific to Moonlight-Switch.  Tried redumping keys, validated all the paths, and worked through the stuff that Avellea tried as well. Only difference I can tell in our configurations is the Atmosphere versions.

Not sure where the issue truely lies here but I did want to chime in that running the Moonlight-Switch nro through the NSP generator site (https://nsp-forwarder.n8.io) worked flawlessly, whereas the same exact files tested in NTON failed to launch, so i'd cautiously assume the issue is with NTON

**Comment by XITRIX** [comment]
Released [Moonlight-Switch NSP generator](https://nsp-forwarder.vercel.app/moonlight), use it to generate forwarders
